source/nrnmatlab.cpp

```diff
@@ -271,4 +272,122 @@ void NrnRef::set(double x) {
 }
 double NrnRef::get() {
     return *(this->ref);
-}
\ No newline at end of file
+}
+
+std::vector<double> get_x3d(Section* sec) {
+    auto result = std::vector<double>(sec->npt3d);
+    for (size_t i = 0; i < sec->npt3d; i++) {
+        result[i] = sec->pt3d[i].x;
+    }
+    return result;
+}
+
+std::vector<double> get_y3d(Section* sec) {
+    auto result = std::vector<double>(sec->npt3d);
+    for (size_t i = 0; i < sec->npt3d; i++) {
+        result[i] = sec->pt3d[i].y;
+    }
+    return result;
+}
+
+std::vector<double> get_z3d(Section* sec) {
+    auto result = std::vector<double>(sec->npt3d);
+    for (size_t i = 0; i < sec->npt3d; i++) {
+        result[i] = sec->pt3d[i].z;
+    }
+    return result;
+}
+
+std::vector<double> get_arc3d(Section* sec) {
+    auto result = std::vector<double>(sec->npt3d);
+    for (size_t i = 0; i < sec->npt3d; i++) {
+        result[i] = sec->pt3d[i].arc;
+    }
+    return result;
+}
+
+std::vector<double> get_d3d(Section* sec) {
+    auto result = std::vector<double>(sec->npt3d);
+    for (size_t i = 0; i < sec->npt3d; i++) {
+        result[i] = sec->pt3d[i].d;
+    }
+    return result;
+}
+
+double lerp(double a, double b, double f) {
+    return a + f * (b - a);
+}
+
+double interp1(std::vector<double> xs, std::vector<double> ys, double v) {
```

Comment:
Are xs and ys sorted?

Reply:
They're ordered in that the ith x value and the ith y value go together, but the shapes are whatever they are and can twist and turn... xs might be increasing as the dendrite bends to the right and then start decreasing as it bends to the left.